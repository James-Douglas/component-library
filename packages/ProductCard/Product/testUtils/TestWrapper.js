import React from 'react';
import ProductProvider from '../../Context/ProductProvider';

const ProductTestWrapper = ({
  // eslint-disable-next-line react/prop-types
  children, product, onDetails = () => {}, onSelect = () => {},
}) => (
  <ProductProvider product={product} onDetails={onDetails} onSelect={onSelect}>
    { children }
  </ProductProvider>
);

export default ProductTestWrapper;
