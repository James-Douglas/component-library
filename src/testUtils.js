// test-utils.js
import {
  render, fireEvent, wait, act,
} from '@testing-library/react';
import ManorProvider from './providers/ManorProvider';

const customRender = (ui, options) => render(ui, { wrapper: ManorProvider, ...options });

// re-export everything
export * from '@testing-library/react';

// override render method
export {
  customRender as render,
  // export other used functions to appease eslint
  fireEvent,
  wait,
  act,
};
