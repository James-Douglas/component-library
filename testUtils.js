import React from 'react';
import {
  render, fireEvent, wait, act,
} from '@testing-library/react';
import ManorProvider from './packages/Provider/ManorProvider';

const ManorTestWrapper = ({ children }) => (
  <ManorProvider>
    {children}
  </ManorProvider>
);

const customRender = (ui, options) => render(ui, { wrapper: ManorTestWrapper, ...options });

// re-export everything
export * from '@testing-library/react';

// override render method
export {
  customRender as render,
  // export other used functions to appease eslint
  fireEvent,
  wait,
  act,
};
