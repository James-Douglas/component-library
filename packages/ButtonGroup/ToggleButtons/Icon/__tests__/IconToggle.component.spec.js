import React from 'react';
import { useTracking } from 'react-tracking';
import { faCheck } from '@fortawesome/free-solid-svg-icons';
import { fireEvent, render } from '../../../../../testUtils';
import IconToggle, { getToggleContent } from '../IconToggle.component';

/* eslint-disable react/prop-types */
describe('getToggleContent()', () => {
  const IconToggleContentContainer = ({
    id, icon, title, description,
  }) => (
    <>{getToggleContent(id, icon, title, description)}</>
  );

  it('renders with props', () => {
    const { getByText, container } = render(
      <IconToggleContentContainer
        id="testing"
        trackingLabel="test-tracking-label"
        icon={faCheck}
        title="test"
        description="test description"
      />,
    );
    expect(getByText('test')).toBeInTheDocument();
    expect(getByText('test description')).toBeInTheDocument();
    expect(container.querySelector('.fa-check')).toBeInTheDocument();
  });
});

describe('IconToggle', () => {
  it('renders with props', () => {
    const { container } = render(
      <IconToggle
        id="test"
        trackingLabel="test-tracking-label"
        value="test"
        title="test icon toggle"
        icon={faCheck}
      />,
    );
    expect(container.innerHTML).toMatchSnapshot();
  });

  describe('interaction tracking', () => {
    it('tracks clicks', () => {
      const { trackEvent } = useTracking();
      const { container } = render(
        <IconToggle
          id="test"
          icon={faCheck}
          trackingLabel="test-tracking-label"
          value="test"
        />,
      );
      const input = container.querySelector('input');
      fireEvent.click(input);
      expect(trackEvent.mock.calls[0][0]).toStrictEqual({
        interaction: {
          ixn_action: 'Click',
          ixn_label: '',
          ixn_object: 'Toggle Buttons',
          ixn_type: 'IconToggle',
          ixn_value: 'test-tracking-label',
        },
      });
    });
  });
});
