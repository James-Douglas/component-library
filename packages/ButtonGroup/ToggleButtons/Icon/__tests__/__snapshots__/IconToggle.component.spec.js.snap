// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IconToggle renders with props 1`] = `"<div class=\\"sc-AxirZ klQxSA\\"><input tabindex=\\"0\\" id=\\"test\\" type=\\"radio\\" name=\\"\\" class=\\"sc-AxiKw botYzr\\" value=\\"test\\"><label for=\\"test\\" class=\\"sc-AxhCb DNJow\\"><div class=\\"sc-AxhUy eQbMTQ\\"><div class=\\"sc-AxgMl duNNlP\\"><div class=\\"sc-AxheI drmA-du\\"><svg aria-hidden=\\"true\\" focusable=\\"false\\" data-prefix=\\"fas\\" data-icon=\\"check\\" class=\\"svg-inline--fa fa-check fa-w-16 fa-4x \\" role=\\"img\\" xmlns=\\"http://www.w3.org/2000/svg\\" viewBox=\\"0 0 512 512\\"><path fill=\\"currentColor\\" d=\\"M173.898 439.404l-166.4-166.4c-9.997-9.997-9.997-26.206 0-36.204l36.203-36.204c9.997-9.998 26.207-9.998 36.204 0L192 312.69 432.095 72.596c9.997-9.997 26.207-9.997 36.204 0l36.203 36.204c9.997 9.997 9.997 26.206 0 36.204l-294.4 294.401c-9.998 9.997-26.207 9.997-36.204-.001z\\"></path></svg></div><div class=\\"sc-Axmtr joSQGB\\"><div class=\\"sc-AxmLO cGWfhR\\"><p class=\\"sc-AxjAm dOroGk\\">test icon toggle</p></div><div class=\\"sc-fzozJi RYGkH\\"><p class=\\"sc-AxjAm dOroGk\\"></p></div></div></div></div></label></div>"`;
